src/mscorlib/src/System/Collections/Generic/Dictionary.cs

```diff
@@ -240,8 +244,10 @@ private struct Entry {
             }
         }
 
-        public bool ContainsKey(TKey key) {
-            return FindEntry(key) >= 0;
+        public bool ContainsKey(TKey key)
+        {
+            TValue unused;
+            return TryGetValue(key, out unused);
```

Comment:
Hmm, that's true, I had not noticed that. I think this is helping elide a copy for the indexer and `TryGetValue` though, since they are now inlined and no longer have to copy when returning. I don't know whether it would be worth it to duplicate all of the hashing/array-searching logic in this method.